vavr-test/build.gradle

```diff
@@ -0,0 +1,41 @@
+apply plugin: 'scala'
+
+description = 'Vavr Test'
+
+sourceSets {
+  main {
+    java {
+      srcDirs += 'src/generated/java'
+    }
+  }
+
+  test {
+    java{
+      srcDirs += 'src/generatedTest/java'
+    }
+  }
+
+  codegen {
+    scala {
+      srcDirs = ['generator']
+    }
+  }
+}
+
+dependencies {
+  compile project(':vavr')
+  testCompile "junit:junit:$junitVersion"
+  testCompile "org.assertj:assertj-core:$assertJVersion"
+
+  codegenCompile "org.scala-lang:scala-library:$scalaVersion"
+  zinc "com.typesafe.zinc:zinc:$zincVersion"
+}
+
+task runScript(type: JavaExec) {
+  inputs.dir file("generator")
+  outputs.dirs file("src/generated"), file("src/generatedTest")
```

Comment:
It would be nice to delete the generated directories before each generator startup to be sure that the old generated files will not remain in the repository (test case: we can rename something in generator).

Also, can we keep the generated code in the `src-gen` folder for a while? This will reduce the changes in PR and help the reviewers. Just asking :)

Reply:
Sure, I'll change both of them. Using `src-gen` is slightly unconventional but there isn't any technical reason it wont work.